### PR TITLE
Common utilities for history parsing

### PR DESCRIPTION
**Pull request by IhateTrains**

### CODE REVIEW

File: ImperatorToCK3/CommonUtils/ContainerField.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+using System.Linq;
+using commonItems;
+
+namespace ImperatorToCK3.CommonUtils {
+	public class ContainerField {
+		public ContainerField(List<string> initialValue) {
+			InitialValue = initialValue;
+		}
+		public List<string> GetValue(Date date) {
+			try {
```
Comment (Idhrendur) [inline ImperatorToCK3/CommonUtils/ContainerField.cs:12]: Exceptions seem like a not great way of handling flow control. I'd prefer to see explicit checks for the things that could go wrong. I'm guessing invalid date and missing value.

File: ImperatorToCK3/CommonUtils/ContainerField.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
```
Comment (Idhrendur) [inline ImperatorToCK3/CommonUtils/ContainerField.cs:1]: Tests

File: ImperatorToCK3/CommonUtils/SimpleField.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+namespace ImperatorToCK3.CommonUtils {
+	public class SimpleField {
+		public SortedDictionary<Date, string> ValueHistory { get; private set; } = new();
+		public string? InitialValue { private get; set; }
+
+		public SimpleField(string? initialValue) {
+			InitialValue = initialValue;
+		}
+		public string? GetValue(Date date) {
+			try {
```
Comment (Idhrendur) [inline ImperatorToCK3/CommonUtils/SimpleField.cs:15]: As above, exceptions seem a bad choice here.

File: ImperatorToCK3/CommonUtils/SimpleField.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
```
Comment (Idhrendur) [inline ImperatorToCK3/CommonUtils/SimpleField.cs:1]: Also tests

File: ImperatorToCK3.UnitTests/CommonUtils/HistoryTests.cs [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+					}
+					1000.1.2 = {
+						culture = cuman
+					}
+				}"
+			);
+
+			var provHistoryFactory = new HistoryFactory(
+				new() {
+					new() { FieldName = "culture", Setter = "culture", InitialValue = null },
```
Comment (Idhrendur) [inline ImperatorToCK3.UnitTests/CommonUtils/HistoryTests.cs:22]: I'd choose initial values other than null here, just to be more explicit.

**Review by Idhrendur — APPROVED** [review]

File: ImperatorToCK3/CommonUtils/ContainerField.cs (hunk 1 above)
Comment (IhateTrains) [reply ImperatorToCK3/CommonUtils/ContainerField.cs]: I switched to checking if `pairsWithEarlierOrSameDate` is empty.

**Pull request merged**